rs/boundary_node/anonymization/client/src/lib.rs

```diff
@@ -11,9 +14,61 @@ use rsa::{
     Pkcs1v15Encrypt, RsaPrivateKey, RsaPublicKey,
 };
 
+#[allow(clippy::disallowed_types)]
```

Comment:
Isn't it disallowed for a reason. Is there no way to do it in an allowed way?

Reply:
Good question, I also asked this initially. The warning was:
```
You should only use an asynchronous lock if you need to .await something while the lock is locked. Usually, this is not necessary, and you should avoid using an asynchronous lock when you can. Asynchronous locks are a lot slower than blocking locks. Please read how to share state effectively across async tasks https://tokio.rs/tokio/tutorial/shared-state. If you are intentionally using an async Mutex, use `#[allow(clippy::disallowed_types)]` to locally disable this check.
```

> If you are intentionally using an async Mutex, use `#[allow(clippy::disallowed_types)]` to locally disable this check.

We really do need an async mutex here because we need to `sleep().await` for the throttle duration (e.g 10 seconds). I _think_ that if we do it with a regular Mutex then basically we'll block a thread for 10s, and since this is meant to run as a background process of `ic-boundary` I didn't want to assume that's OK to do.